### PR TITLE
feat: improve support for allOf, anyOf, oneOf

### PR DESCRIPTION
**Pull request by vwong**

### CODE REVIEW

File: test/e2e/fixtures/openapi3-provider-with-keywords.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                    head:
+                      type: string
+                - type: object
+                  properties:
+                    tail:
+                      type: string
+    Cat:
+      properties:
+        pet_type:
+          const: Cat
```
Comment (vwong) [inline test/e2e/fixtures/openapi3-provider-with-keywords.yaml:62]: Notice this is at the root of `Cat`, while the corresponding one for `Dog` is nested inside `allOf`

File: test/e2e/fixtures/openapi3-provider-with-keywords.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                  - $ref: '#/components/schemas/Dog'
+                discriminator:
+                  propertyName: pet_type
+                required: # this is NOT removed because oneOf needs it
+                  - pet_type
+components:
+  schemas:
+    Pet:
+      type: object
+      required: # this is ignored, because provider returns superset of what consumer uses
```
Comment (vwong) [inline test/e2e/fixtures/openapi3-provider-with-keywords.yaml:29]: This `required` is transformed away as per usual. But the `required` next to `oneOf` is retained.

File: lib/swagger-mock-validator/validate-spec-and-mock/validate-json.ts [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 import {ParsedSpecJsonSchema} from '../spec-parser/parsed-spec';
 
 export const validateJson = (jsonSchema: ParsedSpecJsonSchema, json: any, numbersSentAsStrings?: boolean) => {
     const ajv = new Ajv({
         allErrors: true,
         coerceTypes: numbersSentAsStrings || false,
+        discriminator: true,
         logger: false,
         strictSchema: false,
     });
```
Comment (vwong) [inline lib/swagger-mock-validator/validate-spec-and-mock/validate-json.ts:12]: This is the magic to enable `discriminator` and `draft 2019-09`

File: test/e2e/fixtures/openapi3-provider-with-keywords.yaml (hunk 1 above)
Comment (mefellows) [reply test/e2e/fixtures/openapi3-provider-with-keywords.yaml:62]: It looks to be in both Dog (lines 36-37) and nested within the `allOf` (42-43)?

Screenshot 2023-06-29 at 4 41 35 pm

Comment (vwong) [reply test/e2e/fixtures/openapi3-provider-with-keywords.yaml:60]: Oops, fixed.

**Review by mefellows — APPROVED** [review]
🎉

**Comment by mefellows** [comment]
Happy for you to merge and release this to npm.

**Pull request merged**